src/main/java/com/loqor/client/renderers/PunchGameBlockEntityRenderer.java

```diff
@@ -36,6 +40,7 @@ public void render(T be, float tickDelta, MatrixStack matrices, VertexConsumerPr
 
 		// TODO: Rotate based on state. 
 		// I, @Bug1312, can't visualize this without it taking so much longer than someone who just understands
+		float rot = be.getCachedState().get(TallGameBlock.HORIZONTAL_FACING).asRotation();
```

Comment:
@Loqor are you going to finish this?

Reply:
uhhh sure i can

Reply:
It's just you started and then didn't finish work in an existing PR..

Reply:
i just forgor and got distracted lmao
i just pushed the rest of it, it was one line